src/coreclr/jit/redundantbranchopts.cpp

```diff
@@ -178,15 +178,123 @@ void Compiler::optRelopImpliesRelop(RelopImplicationInfo* rii)
+                    rii->canInfer =
+                        (treeOper == GT_GE) || (treeOper == GT_LE) || (treeOper == GT_GT) || (treeOper == GT_LT);
+                    rii->reverseSense      = ((treeOper == GT_GT) || (treeOper == GT_LT));
+                    rii->canInferFromTrue  = true;
+                    rii->canInferFromFalse = false;
+                    rii->vnRelation        = ValueNumStore::VN_RELATION_KIND::VRK_Unrelated;
+                    break;
+
+                case GT_NE:
+                    rii->canInfer =
+                        (treeOper == GT_GE) || (treeOper == GT_LE) || (treeOper == GT_GT) || (treeOper == GT_LT);
+                    rii->reverseSense      = ((treeOper == GT_GE) || (treeOper == GT_LE));
+                    rii->canInferFromTrue  = false;
+                    rii->canInferFromFalse = true;
+                    rii->vnRelation        = ValueNumStore::VN_RELATION_KIND::VRK_Unrelated;
+                    break;
+
+                case GT_LE:
+                    rii->canInfer =
+                        (treeOper == GT_NE) || (treeOper == GT_EQ) || (treeOper == GT_GE) || (treeOper == GT_LT);
+                    rii->reverseSense      = ((treeOper == GT_NE) || (treeOper == GT_GE));
+                    rii->canInferFromFalse = true;
+                    rii->canInferFromTrue  = false;
+                    rii->vnRelation        = ValueNumStore::VN_RELATION_KIND::VRK_Unrelated;
+                    break;
+
+                case GT_GT:
+                    rii->canInfer =
+                        (treeOper == GT_NE) || (treeOper == GT_EQ) || (treeOper == GT_GE) || (treeOper == GT_LT);
+                    rii->reverseSense      = ((treeOper == GT_EQ) || (treeOper == GT_LT));
+                    rii->canInferFromFalse = false;
+                    rii->canInferFromTrue  = true;
+                    rii->vnRelation        = ValueNumStore::VN_RELATION_KIND::VRK_Unrelated;
+                    break;
+
+                case GT_GE:
+                    rii->canInfer =
+                        (treeOper == GT_NE) || (treeOper == GT_EQ) || (treeOper == GT_GT) || (treeOper == GT_LE);
+                    rii->reverseSense      = ((treeOper == GT_NE) || (treeOper == GT_LE));
+                    rii->canInferFromFalse = true;
+                    rii->canInferFromTrue  = false;
+                    rii->vnRelation        = ValueNumStore::VN_RELATION_KIND::VRK_Unrelated;
+                    break;
+
+                case GT_LT:
+                    rii->canInfer =
+                        (treeOper == GT_NE) || (treeOper == GT_EQ) || (treeOper == GT_GT) || (treeOper == GT_LE);
+                    rii->reverseSense      = ((treeOper == GT_EQ) || (treeOper == GT_GT));
+                    rii->canInferFromFalse = false;
+                    rii->canInferFromTrue  = true;
+                    rii->vnRelation        = ValueNumStore::VN_RELATION_KIND::VRK_Unrelated;
+                    break;
```

Comment:
@AndyAyersMS nit: in my attempt to do something very similar I ended up with a sort of a table, like this:
![image](https://user-images.githubusercontent.com/523221/184985518-8a6a07e5-63a4-425f-8b17-c56e5c8ccd67.png)

Reply:
Yeah, there's probably some refactoring like this that makes sense.